### PR TITLE
[FIX] point_of_sale: fix session report payment difference

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
If you use a non cash payment method, with "identify customer" option
and register a payment difference at the closing of the session. The
difference is not taken into account in the session report.

Steps to reproduce:
- Activate "identify customer" option on the bank payment method
- Open a session
- Add a product to the order
- Pay the order with the bank payment method
- Close the session with a payment difference
- Print the session report
- Open the report, at the bottom you should see the payment difference
  but it is not the case

opw-3597291
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by robinengels** [comment]
check behavior with normal cash different loss to compare

File: addons/point_of_sale/models/report_sale_details.py [hunk 1]
```diff
@@ -126,23 +126,26 @@ def get_sale_details(self, date_start=False, date_stop=False, config_ids=False,
...
+                            payment['money_counted'] = payment['final_count'] + payment['money_difference']
+                            payment['cash_moves'] = []
+                            if is_profit:
+                                move_name = 'Difference observed during the counting (Profit)'
+                                payment['cash_moves'] = [{'name': move_name, 'amount': payment['money_difference']}]
+                            elif is_loss:
+                                move_name = 'Difference observed during the counting (Loss)'
+                                payment['cash_moves'] = [{'name': move_name, 'amount': payment['money_difference']}]
+                            payment['count'] = True
+                            break
```
Comment (caburj) [inline addons/point_of_sale/models/report_sale_details.py:148]:
```suggestion
                        account_move = next(self.env['account.move'].search([("ref", "=", ref_value)], limit=1))
                        if account_move:
                            payment_method = self.env['pos.payment.method'].browse(payment['id'])
                            is_loss = any(l.account_id == payment_method.journal_id.loss_account_id for l in account_move.line_ids)
                            is_profit = any(l.account_id == payment_method.journal_id.profit_account_id for l in account_move.line_ids)
                            payment['final_count'] = payment['total']
                            payment['money_difference'] = -account_move.amount_total if is_loss else account_move.amount_total
                            payment['money_counted'] = payment['final_count'] + payment['money_difference']
                            payment['cash_moves'] = []
                            if is_profit:
                                move_name = 'Difference observed during the counting (Profit)'
                                payment['cash_moves'] = [{'name': move_name, 'amount': payment['money_difference']}]
                            elif is_loss:
                                move_name = 'Difference observed during the counting (Loss)'
                                payment['cash_moves'] = [{'name': move_name, 'amount': payment['money_difference']}]
                            payment['count'] = True
```

If you only want the first difference account move, I prefer this if it works.

Comment (robinengels) [reply addons/point_of_sale/models/report_sale_details.py]: We can't use `next`, but yes we actually only need one account_move.

File: addons/point_of_sale/models/report_sale_details.py [hunk 2]
```diff
@@ -126,23 +126,26 @@ def get_sale_details(self, date_start=False, date_stop=False, config_ids=False,
...
```
Comment (caburj) [inline addons/point_of_sale/models/report_sale_details.py:148]: remove this `break`

**Comment by robinengels** [comment]
@caburj
